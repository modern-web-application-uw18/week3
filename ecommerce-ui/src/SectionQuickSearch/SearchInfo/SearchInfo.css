.quickSearch .title {
  font-family: 'Play', sans-serif;
}
