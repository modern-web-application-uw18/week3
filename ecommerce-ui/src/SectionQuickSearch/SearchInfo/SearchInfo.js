import React,{Component} from 'react';
import PropTypes from 'prop-types';

import './SearchInfo.css';

export default class SearchInfo extends Component {

    render() {
      const { houseAmount, description } = this.props.search;
        return (
          <div className="quickSearch">
            <div className="title">
              {houseAmount >= 2000 ? '2000+' : houseAmount} airbnbs available
            </div>
            <div className="text">
              {description}
            </div>
          </div>
        );
    }
}

SearchInfo.propTypes = {
  search: PropTypes.shape({
    houseAmount: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired
  })
}

SearchInfo.defaultProps = {
  search: {
    houseAmount: 0,
    description: 'test_description'
  }
}
