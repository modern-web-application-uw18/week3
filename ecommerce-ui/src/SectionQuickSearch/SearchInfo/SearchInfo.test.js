import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';

import SearchInfo from './SearchInfo.js';

describe('SearchInfo', () => {
  it('Smoke Test', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchInfo />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Match Snapshot', () => {
    const component = TestRenderer.create(<SearchInfo />);
    expect(component.toJSON()).toMatchSnapshot();
  });

})
