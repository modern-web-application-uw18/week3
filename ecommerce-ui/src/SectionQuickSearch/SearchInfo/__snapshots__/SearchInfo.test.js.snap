// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInfo Match Snapshot 1`] = `
<div
  className="quickSearch"
>
  <div
    className="title"
  >
    0
     airbnbs available
  </div>
  <div
    className="text"
  >
    test_description
  </div>
</div>
`;
